addons/point_of_sale/models/account_journal.py

```diff
@@ -27,3 +28,17 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
     def onchange_type(self):
         if self.type not in ['bank', 'cash']:
             self.journal_user = False
+
+    @api.multi
+    def write(self, vals):
+        for journal in self:
+            if 'active' in vals and self.env['pos.session'].search_count([('state', '!=', 'closed'), '|', ('config_id.journal_id', '=', journal.id), ('config_id.journal_ids', 'in', journal.id)]):
+                raise UserError(_("You cannot delete Journals that are used by active PoS sessions."))
+        return super(AccountJournal, self).write(vals)
+
+    @api.multi
+    def unlink(self):
+        for journal in self:
+            if self.env['pos.session'].search_count([('state', '!=', 'closed'), '|', ('config_id.journal_id', '=', journal.id), ('config_id.journal_ids', 'in', journal.id)]):
+                raise UserError(_("You cannot delete Journals that are used by active PoS sessions."))
```

Comment:
same as above, the name of the journal and the session should be in the error message